lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -242,6 +242,63 @@ static Value createTMTensorScanOp(
   return scanOp->getResult(0);
 }
 
+// Utility function to create a TMTensor::SortOp.
+static FailureOr<SmallVector<Value, 2>>
+createTMTensorSortOp(PatternRewriter &rewriter, Location sortOpLoc,
+                     llvm::ArrayRef<Value> operands,
+                     llvm::ArrayRef<Type> elementTypes, int64_t dimension,
+                     bool isStable, bool isDescending) {
+  // Step 1. Create TMTensor::SortOp structure.
+  SmallVector<Type> sortResultTypes;
+  for (Value val : operands) {
+    sortResultTypes.push_back(val.getType());
+  }
+  ValueRange inputs;
+  auto sortOp = rewriter.create<TMTensor::SortOp>(
+      sortOpLoc, sortResultTypes, inputs, operands,
+      rewriter.getI64IntegerAttr(dimension));
+
+  // Step 2. Add two arguments for each element type in the SortOp's block.
+  Region *body = &sortOp.getRegion();
+  Block *block = rewriter.createBlock(body);
+  Location loc = body->getLoc();
+  for (Type elementType : elementTypes) {
+    block->addArguments({elementType, elementType},
+                        SmallVector<Location, 2>(2, loc));
+  }
+
+  // Step 3. Create comparison op which will be used as the sorting predicate.
+  Value compareOp;
+  if (auto intType = elementTypes[0].dyn_cast<mlir::IntegerType>()) {
+    // Case for using arith::CmpIOp.
+    arith::CmpIPredicate ge = arith::CmpIPredicate::sge;
+    arith::CmpIPredicate le = arith::CmpIPredicate::sle;
+    if (intType.isUnsignedInteger()) {
+      ge = arith::CmpIPredicate::uge;
+      le = arith::CmpIPredicate::ule;
+    }
+    arith::CmpIPredicate predicate = isDescending ? ge : le;
+    compareOp = rewriter.create<arith::CmpIOp>(
+        loc, predicate, block->getArgument(0), block->getArgument(1));
+  } else if (elementTypes[0].isa<mlir::FloatType>()) {
+    // Case for using arith::CmpFOp.
+    arith::CmpFPredicate predicate =
+        isDescending ? arith::CmpFPredicate::OGE : arith::CmpFPredicate::OLE;
+    compareOp = rewriter.create<arith::CmpFOp>(
+        loc, predicate, block->getArgument(0), block->getArgument(1));
+  } else {
+    return rewriter.notifyMatchFailure(
+        sortOpLoc, "Only Integer and Floating element type expected.");
+  }
+
+  // Step 4. Create yield op for yielding the sorting predicate.
+  rewriter.create<TMTensor::YieldOp>(loc, compareOp);
+  SmallVector<Value, 2> sortOpValues;
+  sortOpValues.push_back(sortOp.getResult(0));
+  sortOpValues.push_back(sortOp.getResult(1));
```

Comment:
The function `createTMTensorSortOp` does not limit the number of inputs, so the number of outputs should also not be limited. The result type should change to `FailureOr<SmallVector<Value>>` and here I would just do `return sortOp.getResults()` (you might need to add a `SmallVector` constructor)